src/librustc_errors/lib.rs

```diff
@@ -350,6 +362,8 @@ pub struct HandlerFlags {
 
 impl Drop for Handler {
     fn drop(&mut self) {
+        self.emit_stashed_diagnostics();
```

Comment:
Same idea here; better safe than sorry.

Reply:
I think only this one is necessary, but I'm ok with both.